### ISSUE TITLE
Tree View: File tree collapses file structure on refresh

### ISSUE THREAD

**Issue by jasonhawkharris** [sourcegraph/sourcegraph#58409]

**Comment by jasonhawkharris** [comment]
CC: @jdpleiness

**Comment by jasonhawkharris** [comment]
https://www.loom.com/share/cc74d97ea6cb492d9e67873247c25d53?sid=0e88338a-d4e2-4cfc-9267-4d18919093c1

**Comment by jasonhawkharris** [comment]
When utilizing console.log to track the data object within the [TreeView component](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/wildcard/src/components/Tree/react-accessible-treeview/index.tsx?L808), I observe that the length of the data array remains constant while navigating deeper into the file structure. However, upon refreshing, the items in the data array reduce to the number of items within the immediate parent folder of the currently displayed file in the blob view. To see this, follow these steps:

1) Add a `useEffect` with a `console.log` statement [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/wildcard/src/components/Tree/react-accessible-treeview/index.tsx?L854) to log the [data object at this point](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/wildcard/src/components/Tree/react-accessible-treeview/index.tsx?L810).
2) Open the browser and the browser console.
3) In the development environment, navigate to the TreeView of the [sourcegraph/sourcegraph repo](https://sourcegraph.com/github.com/sourcegraph/sourcegraph). Take note of the initial length of `data` in the console (87).
4) Click through the file structure in the following order, observing that the data length remains unchanged:
    - `client/`
    - `wildcard/`
    - `src/`
    - `components/`
    - `Tree/`
    - `Tree.tsx`
5) Upon reaching the blob view of `Tree.tsx`, notice that the file structure is not collapsed, and it follows the clicked path. Refresh the page and observe that the length of `data` is now significantly smaller (9). Additionally, the first item in the array is the absolute path to the current file. This can be disorienting for the user.
6) Using the file tree, backtrack to the root of the repository, and observe the changing length of the data array at each step.

To address this issue, we should allow the expanded file structure to persist (remain uncollapsed) on refresh. This behavior seems to be an attempt to optimize data fetching when visiting the blob view. However, it is recommended to maintain the full expanded tree at every step during navigation, similar to how GitHub code search operates.

**Comment by vovakulikov** [comment]
@jasonhawkharris Yes, as I remember, we don't want to load all file structures (which means load all files in the repository) if you just want to see one file, I think we resolve the file path and its parent folder by URL.

I'm not sure 100%, but possible fix for this problem should be very simple: we should preserve the [`initialPathFile`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/repo/RepoRevisionSidebar.tsx?L66:5) somewhere in a persisted store (I think we should go with a session store; the temporary store is too much, and it's async, local-storage also seems to be too much and could be annoying, session storage should do the work here since it will be reset next time when we open new page)

So maybe this would be enough. (I'm not sure about the session's storage key. Maybe we should be more specific than just repoName, maybe add commit) 

```tsx
const [initialFilePath, setInitialFilePath] = useSessionStorage<string>(`file-tree-sessions-storage-${repoName}`, props.filePath)
```
